src/target/nrf54l.c

```diff
@@ -0,0 +1,166 @@
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+#include "adiv5.h"
+
+#define NRF54L_FICR_INFO_RAM  0x00ffc328U
+#define NRF54L_FICR_INFO_RRAM 0x00ffc32cU
+
+#define NRF54L_UICR 0x00ffd000U
+
+#define NRF54L_RRAMC_READY          0x5004b400U
+#define NRF54L_RRAMC_CONFIG         0x5004b500U
+#define NRF54L_RRAMC_ERASE_ERASEALL 0x5004b540U
+
+#define NRF54L_CTRL_AP_IDR_VALUE        0x32880000U
+#define NRF54L_CTRL_AP_RESET            ADIV5_AP_REG(0x00U)
+#define NRF54L_CTRL_AP_ERASEALL         ADIV5_AP_REG(0x04U)
+#define NRF54L_CTRL_AP_ERASEALLSTATUS   ADIV5_AP_REG(0x08U)
+#define NRF54L_CTRL_AP_APPROTECT_STATUS ADIV5_AP_REG(0x14U)
+
+static bool rram_erase(target_flash_s *f, target_addr_t addr, size_t len)
+{
+	(void)f;
+	(void)addr;
+	(void)len;
+	// RRAM doesn't need to be erased before being written, so we just return ok.
+	return true;
+}
+
+static bool rram_write(target_flash_s *f, target_addr_t dest, const void *src, size_t len)
+{
+	target_mem32_write(f->t, dest, src, len);
+	return true;
+}
+
+static bool rram_prepare(target_flash_s *f)
+{
+	target_mem32_write32(f->t, NRF54L_RRAMC_CONFIG, 1);
+	return true;
+}
+
+static bool rram_done(target_flash_s *f)
+{
+	target_mem32_write32(f->t, NRF54L_RRAMC_CONFIG, 0);
+	return true;
+}
+
+static bool rram_mass_erase(target_s *const t, platform_timeout_s *const print_progess)
+{
+	target_mem32_write32(t, NRF54L_RRAMC_ERASE_ERASEALL, 1);
+
+	uint32_t ready;
+	do {
+		ready = target_mem32_read32(t, NRF54L_RRAMC_READY);
+		target_print_progress(print_progess);
+	} while (ready == 0);
+
+	return true;
+}
+
+static void add_rram(target_s *t, uint32_t addr, size_t length)
+{
+	target_flash_s *f = calloc(1, sizeof(*f));
+	if (!f) { /* calloc failed: heap exhaustion */
+		DEBUG_ERROR("calloc: failed in %s\n", __func__);
+		return;
+	}
+
+	f->start = addr;
+	f->length = length;
+	f->blocksize = 1024U;
+	f->writesize = 1024U;
+	f->erase = rram_erase;
+	f->write = rram_write;
+	f->prepare = rram_prepare;
+	f->done = rram_done;
+	f->erased = 0xff;
+	target_add_flash(t, f);
+}
+
+bool nrf54l_probe(target_s *target)
+{
+	DEBUG_INFO("nrf54l_probe:\n");
+
+	adiv5_access_port_s *ap = cortex_ap(target);
+
+	if (ap->dp->version < 2U)
+		return false;
+
+	switch (ap->dp->target_partno) {
+	case 0x1c0:
```

Comment:
Please name this constant (having suffixed it to make it unsigned) - something like `ID_NRF45L` would fit with the nomenclature for part numbers.